tensorflow/compiler/tf2tensorrt/common/utils.h

```diff
@@ -32,9 +32,35 @@ std::tuple<int, int, int> GetLoadedTensorRTVersion();
 
 #if GOOGLE_CUDA && GOOGLE_TENSORRT
 
+#include "tensorflow/core/platform/errors.h"
 #include "tensorflow/core/platform/logging.h"
 #include "third_party/tensorrt/NvInfer.h"
 
+#define TRT_EXPECT(x)                                  \
+  if (!(x)) {                                          \
+    return errors::Internal(__FILE__, ":", __LINE__,   \
+                            " pre condition failure"); \
+  }
+
+#define TRT_ENSURE(x)                                   \
+  if (!(x)) {                                           \
+    return errors::Internal(__FILE__, ":", __LINE__,    \
+                            " post condition failure"); \
+  }
+
+#define TRT_ENSURE_OK(x)                                 \
+  if (!x.ok()) {                                         \
+    return errors::Internal(x.status().error_message()); \
+  }
+
+#define TRT_ENSURE_PTR_OK(x)                                \
+  if (!x.ok()) {                                         \
+    return errors::Internal(x.status().error_message()); \
+  } else if (*x == nullptr) {                            \
+    return errors::Internal(__FILE__, ":", __LINE__,     \
+                            "pointer had null value");   \
+  }
```

Comment:
TRT_ENSURE_PTR_OK is if what is in the template paramter for `StatusOr<>` is a pointer, like` StatusOr<SomeObj*>`. I added this due to recommendation from https://abseil.io/docs/cpp/guides/status . For usage with TRTNetworkBuilder class in `layer_utils.h`, this is repetetive because I already checked `nullptr` case before returning, but still it can't hurt.

Reply:
Basically I used it when I was consuming TRTNetworkBuilder API, but internally to a class implementation, if there's an existing nullptr check, I might not have used it.

Reply:
The macros here are for internal consistency check and ideally they will never fail, right?
Before you added TRT_ENSURE_PTR_OK and only use it for build->Slice, I was fine with just using TRT_ENSURE_OK with the assumption that all routines in our implementation that return StatusOr<T*> either return an error status or a non-null ptr. I actually prefer rolling back to that status.

However, after you added TRT_ENSURE_PTR_OK and only use it for build->Slice, I start to question how we can  choose a macro between these two when writing new code? We should document these two macros with such information, if we really need both. I see build->Slice either return an error status or a non-null pointer.
I prefer to keep only one version. But if you prefer to keep both, we need to document how to choose one over another. Currently, the logic you use to choose a version is something like this: "if the StatusOr<T*> is produced by member functions in the same class, and the implementation ensures that it either return an error status or a non-null pointer, we use TRT_ENSURE_OK"?

Reply:
I changed all StatusOr<T*> checks to use `TRT_ENSURE_PTR_OK` for consistency, as that enforces correct checks